src/tigerbeetle/main.zig

```diff
@@ -348,8 +385,72 @@ fn replica_release_execute(replica: *Replica, release: vsr.Release) noreturn {
         @panic("release_execute: binary missing required version");
     }
 
-    // TODO(Multiversioning) Exec into the new release.
-    unreachable;
+    // This is a custom build for the epoch of multiversioning. The binary_path will always be
+    // argv[0] since this binary would never have been executed by a user directly, and our caller
+    // will provide an absolute path there.
+    switch (builtin.os.tag) {
+        .macos, .linux => {
+            var binary_path = std.os.argv[0];
+            assert(std.fs.path.isAbsolute(std.mem.span(binary_path)));
+
+            // We can pass through our env and args as-is to exec. We have to manipulate the types
+            // here somewhat: they're cast in start.zig and we can't access `argc_argv_ptr`
+            // directly.
+            // process.zig does the same trick in execve().
+            const cast_args: [*:null]const ?[*:0]const u8 = @ptrCast(std.os.argv.ptr);
+            const cast_envp: [*:null]const ?[*:0]const u8 = @ptrCast(std.os.environ.ptr);
+
+            std.log.info("replica_release_execute: executing '{s}'...\n", .{binary_path});
+            std.os.execveZ(binary_path, cast_args, cast_envp) catch @panic("execveZ failed");
+            unreachable;
+        },
+        .windows => {
+            var binary_path = replica_release_execute_args.?.next().?;
+            assert(std.fs.path.isAbsolute(binary_path));
```

Comment:
> Is the global replica_release_execute_args needed by "real" multiversioning too?

Nope, there it's handled inside `multiversioning.zig`.

I've removed the branch for now, at the cost of a little extra memory on Linux / macOS, but since this is a transitional release only I think that's OK!